src/map.rs

```diff
@@ -457,6 +457,37 @@ where
         }
     }
 
+    /// Returns a reference to the value with the given `hash` and which
+    /// satisfies the equality function passed.
+    ///
+    /// This function is similar to [`Self::get()`], but provides a shortcut
+    /// for lookups where instantiation of keys is expensive. For instance, if
```

Comment:
Wouldn't this use case be better served by the [`Equivalent`](https://docs.rs/hashbrown/latest/hashbrown/trait.Equivalent.html) trait? `get_by_hash` is useful but I think serves a different use case, such as supporting unhashable types or precomputed hashes. For that I wonder if it would be better to expose a lower-level type such as hashbrown's [`HashTable`](https://docs.rs/hashbrown/latest/hashbrown/struct.HashTable.html).

Reply:
Great point! I've switched to using an `Equivalent` trait instead, so we don't need an additional method. This also leaves fewer opportunities for consumers to misuse the API.